Made the DefaultGroovyMethodsTest file more groovy like.

russellsanborn opened the pull request:
* Removed unnecessary semicolons and `public` modifiers.
* replaced java style for loops with groovy `eachWithIndex` closures
* instantiating maps and lists with `[]` and `[:]` instead of using `new ArrayList`, `Arrays.asList`, or `new LinkedHashMap`

paulk-asert commented:
The first three bullet points are non-controversial changes - the file was originally a Java file and wasn't refactored much when changed to Groovy. The null on the LHS probably requires more discussion on the mailing list. There are places where we try to hide NPEs and other places we don't. We value consistency across all cases ahead of commutativity (consistency between LHS and RHS).

Splitting into two PRs or at least two commits would make it easier to progress the non-controversial changes.

russellsanborn commented:
@paulk-asert I updated this PR to only have the non-controversial changes and updated the bullet points.

I opened a new PR with the changes to `DefaultGroovyMethods` boolean operations:
https://github.com/apache/groovy/pull/636

paulk-asert commented:
Merged, thanks.
